docs/sdks/rust/index.md

```diff
@@ -177,29 +334,114 @@ trait DbContext {
 
 Subscribe to queries by calling `ctx.subscription_builder()` and chaining configuration methods, then calling `.subscribe(queries)`.
 
-#### Callback `on_applied`
+##### Callback `on_applied`
+
+```rust
+impl SubscriptionBuilder {
+    fn on_applied(self, callback: impl FnOnce(&SubscriptionEventContext)) -> Self;
+}
+```
+
+Register a callback to run when the subscription is applied and the matching rows are inserted into the client cache.
+
+##### Callback `on_error`
 
 ```rust
 impl SubscriptionBuilder {
-    fn on_applied(self, callback: impl FnOnce(&EventContext)) -> Self;
+    fn on_applied(self, callback: impl FnOnce(&ErrorContext, spacetimedb_sdk::Error)) -> Self;
 }
 ```
 
-Register a callback to run when the subscription is applied and the matching rows are inserted into the client cache. The [`EventContext`](#type-eventcontext) passed to the callback will have `Event::SubscribeApplied` as its `event`.
+Register a callback to run if the subscription is rejected or unexpectedly terminated by the server. This is most frequently caused by passing an invalid query to [`subscribe`](#method-subscribe).
+
 
-#### Method `subscribe`
+##### Method `subscribe`
 
 ```rust
 impl SubscriptionBuilder {
     fn subscribe(self, queries: impl IntoQueries) -> SubscriptionHandle;
 }
 ```
 
-Subscribe to a set of queries. `queries` should be an array or slice of strings.
+Subscribe to a set of queries. `queries` should be a string or an array, vec or slice of strings.
```

Comment:
link to sql docs

Reply:
Added in a paragraph directly after this one.